### PR TITLE
Flagging sensitive params

### LINKED ISSUE #3379: Update all SaaS config connector params with sensitive flag

### PR DESCRIPTION
**Pull request by galvana**
Closes #3379 

### Code Changes

* [ ] Added `sensitive` flag to the SaaS config's `connector_param` field
* [ ] Updated the fields in the DB schemas with the sensitive flag
* [ ] Defined `title` values for select DB schema fields to fix the "human readable" label for the UI

### Steps to Confirm

* [ ] This is not going to be visible to the user in this iteration, this is just preparation for future front-end work

### Pre-Merge Checklist

* [ ] All CI Pipelines Succeeded
* Documentation:
  * [ ] documentation complete, [PR opened in fidesdocs](https://github.com/ethyca/fidesdocs/pulls)
  * [ ] documentation [issue created in fidesdocs](https://github.com/ethyca/fidesdocs/issues/new/choose)
* [ ] Issue Requirements are Met
* [ ] Relevant Follow-Up Issues Created
* [ ] Update `CHANGELOG.md`
* [ ] For API changes, the [Postman collection](https://github.com/ethyca/fides/blob/main/docs/fides/docs/development/postman/Fides.postman_collection.json) has been updated

### CODE REVIEW

**Comment by daveqnet** [comment]
@galvana, before I add any review comments, I'll say that a bunch of them will be borderline, subjective opinions on what's considered "sensitive".

There are fields which are definitely sensitive, fields which are definitely not sensitive, and everything else lives in a grey area.

As a secure design principle, my first instinct would be to default all the grey area fields to sensitive, but if we put up too many hurdles for admins they're just going to turn off hiding and make everything insecure! It's a finely balanced UX problem. 🙂

For that reason I'm going to try to err a little more on the side of convenience than security in my review (heresy coming from me, I know).

File: data/saas/config/yotpo_loyalty_config.yml [hunk 1]
```diff
@@ -14,10 +14,12 @@ saas_config:
       label: GUID
     - name: api_key
       label: API Key
+      sensitive: True
```
Comment (daveqnet) [inline data/saas/config/yotpo_loyalty_config.yml:17]: What is the API key field used for? As far as I can tell, Yotpo auth only needs the store ID & secret key?

https://core-api.yotpo.com/reference/authentication

File: src/fides/api/schemas/connection_configuration/connection_secrets_fides.py [hunk 2]
```diff
@@ -10,8 +12,8 @@ class FidesConnectorSchema(ConnectionConfigSecretsSchema):
     """Schema to validate the secrets needed to connect to a remote Fides"""
 
     uri: str
-    username: str
-    password: str
+    username: str = Field(sensitive=True)
+    password: str = Field(sensitive=True)
```
Comment (daveqnet) [inline src/fides/api/schemas/connection_configuration/connection_secrets_fides.py:16]: We can connect to remote fides instances? Is this brand new - I can't find it in the fides 2.14.0 admin-ui add connector page?

**Review by daveqnet — COMMENTED** [review]
Hi @galvana,

Aside from my inline comments, is the timescale connector missing?

You'll also have to get a dev to review your code.

Thanks!

File: data/saas/config/yotpo_loyalty_config.yml (hunk 1 above)
Comment (galvana) [reply data/saas/config/yotpo_loyalty_config.yml:17]: The Yotpo Loyalty API doesn't allow programatic erasures through any of the documented API endpoints. We bypassed this by inspecting the network traffic when a user is deleted from the Yotpo Loyalty dashboard and reproducing the call in the connector config. This endpoint requires the API key that is found at https://loyalty-app.yotpo.com/general-settings. Might not be the best approach but it works 😬.

File: data/saas/config/wunderkind_config.yml [hunk 3]
```diff
@@ -3,10 +3,11 @@ saas_config:
   name: Wunderkind
   type: wunderkind
   description: A schema representing the Wunderkind connector for Fides
-  version: 0.0.1
+  version: 0.0.2
 
   connector_params:
     - name: website_id
+      label: Website ID
```
[the remark this answers is not in the document]
Comment (galvana) [reply data/saas/config/wunderkind_config.yml:10]: It looks like the `/ccpa` endpoint is public and the unique identifier is just the `website_id`.

File: data/saas/config/segment_config.yml [hunk 4]
```diff
@@ -12,8 +12,11 @@ saas_config:
       default_value: profiles.segment.com
     - name: workspace
     - name: access_token
+      sensitive: True
     - name: namespace_id
+      label: Namespace ID
     - name: access_secret
+      sensitive: True
```
[the remark this answers is not in the document]
Comment (galvana) [reply data/saas/config/segment_config.yml:19]: There's only one mention of `Access Secret` in the docs but it refers to functionality for a higher-tiered service that we don't have access to test https://segment.com/docs/unify/profile-api/. It looks like it should still be considered a secret.

**Comment by galvana** [comment]
> Hi @galvana,
> 
> Aside from my inline comments, is the timescale connector missing?
> 
> You'll also have to get a dev to review your code.
> 
> Thanks!

Hi @daveqnet, the Timescale connector uses the same schema as Postgres so we're covered there.

**Review by daveqnet — APPROVED** [review]
Reviewed and approved from a security perspective, with my thanks, @galvana!

File: src/fides/api/schemas/connection_configuration/connection_secrets.py [hunk 5]
```diff
@@ -12,7 +13,9 @@
 class ConnectionConfigSecretsSchema(BaseModel, abc.ABC):
     """Abstract Base Schema for updating Connection Configuration Secrets"""
 
-    url: Optional[str] = None  # User can always specify the URL directly
+    url: Optional[str] = Field(
+        None, title="URL", sensitive=True
```
Comment (TheAndrewJackson) [inline src/fides/api/schemas/connection_configuration/connection_secrets.py:17]: I didn't know you can add arbitrary fields to the pydantic `Field` class.

**Review by TheAndrewJackson — APPROVED** [review]
💥 Looks good!

**Pull request merged**